### ISSUE TITLE
feat: make position of `<style>` tags configurable in `hydrateDocument()`

### ISSUE THREAD

**Issue by mhoritani** [ionic-team/stencil#5915]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already include this feature request, without success.

### Describe the Feature Request

It would be helpful if `hydrateDocument()` could be configured as to where generated inline <style> tags will be placed.

### Describe the Use Case

When the document is hydrated, the inline `<style>` tags created for the Stencil components are placed at the top of the `<head>` section. This placement can delay the loading of `<link>` tags, such as those with a `rel="preconnect"` value, which are crucial for performance optimization. Ideally, the placement of these styles should be configurable to avoid such issues.

**Sample input:**

```html
<!doctype html>
<html lang="en">
    <head>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>Document</title>
        <link rel="preconnect" href="https://some-url.com" />
    </head>
    <body>
        <my-component></my-component>
    </body>
</html>
```

**Output:**
```html
<!doctype html>
<html lang="en" data-stencil-build="vuogzrh7" class="hydrated">
    <head>
        <meta charset="UTF-8" />

        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>Document</title>
        <style sty-id="sc-my-component">
            /*!@:host*/
            .sc-my-component-h {
                display: inline-block;
                border: 1px solid red;
                padding: 5px;
            }
        </style>
        <link rel="preconnect" href="https://some-url.com" />
    </head>
    <body>
        <my-component class="sc-my-component-h hydrated" s-id="1"
            ><!--r.1-->
            <div class="sc-my-component" c-id="1.0.0.0">
                <!--t.1.1.1.0-->My Component
            </div></my-component
        >
    </body>
</html>
```

### Describe Preferred Solution

Make `hydrateDocument()` configurable as to where generated style tags are placed.

### Related Code

A well-documented reproduction of the use-case/issue can be found here:
https://github.com/mhoritani/stencil-repro-072024

**Comment by christian-bromann** [comment]
Thanks for reporting @mhoritani , I think changing the order makes totally sense. Any contributions are much appreciated!

**Comment by mhoritani** [comment]
Happy to have a look @christian-bromann. Any chance of pointing us in the right direction?

**Comment by christian-bromann** [comment]
@mhoritani I pulled the issue into the sprint to take a look at.

For future reference, this behavior was introduced in https://github.com/ionic-team/stencil/commit/05d242d5d6ba558896495df3f7b923ca109c70d9 unfortunately with no further explanation on the reason.

**Comment by christian-bromann** [comment]
A fix for this bug has been published in Stencil [`v4.21.0`](https://github.com/ionic-team/stencil/releases/tag/v4.21.0)!
